Documentation/Migration.md

```diff
@@ -0,0 +1,11 @@
+# Migrating Between Versions
+
+## Migrating from Aardvark 3.x / CoreAardvark 2.x
+
+The biggest change in the Aardvark 4.0 family is the reorganization of files between frameworks. This release introduced two new frameworks: AardvarkMailUI and AardvarkLoggingUI. Both of these frameworks were released with a version number of 1.0. You will likely need to update your dependencies to include one or more of these new frameworks, depending which components of Aardvark you were previously using.
+
+In addition, there are a handful of breaking changes in the API:
+
+* `ARKEmailAttachment` was renamed to `ARKBugReportAttachment`.
+* `ARKEmailBugReporter.attachesViewHierarchyDescriptionWithScreenshot` was renamed to `ARKEmailBugReporter.attachesViewHierarchyDescription`. This also represents a change in when the view hierarchy is attached (it no longer depends on the screenshot also being attached).
+* The initializer for `ARKLogMessage` now takes an additional `parameters` argument. This allows for attaching a persisted string dictionary to the log message.
```

Comment:
😅 can't we make this change invisible by having a default argument or convenience initializer?

Reply:
Yeah, I suppose we could add a convenience initializer (it's Objective-C, so default argument isn't an option). There's currently only two initializers (one with all arguments and one that uses the current date). I feel like if we want to go the route of providing convenience initializers with arguments omitted, we should be consistent and do it for `image` and `userInfo` as well, and maybe `type`. Doing it for only `parameters` is nice in that it keeps it from being a breaking change, but it feels like it leaves the API in a weird place.

In any case, I feel like creating a log message manually is an edge case. The one case I could see it making sense is if you subclassed `ARKLogMessage`, but then you only have one place to update the initializer anyway.

Thoughts?